### PR TITLE
feat: new snapshot architecture on the cmdline

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

**Comment by abentkamp** [comment]
There are a couple of restrictions of emscripten pthreads listed here,
which might cause this:

https://emscripten.org/docs/porting/pthreads.html

Unfortunately, I dont think I will have the time to look into this in the
near future.

Sebastian Ullrich ***@***.***> schrieb am Sa., 23. Dez. 2023,
5:06 PM:

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

File: src/Lean/Elab/Frontend.lean [hunk 1]
```diff
@@ -81,23 +81,17 @@ end Frontend
 
 open Frontend
```
Comment (Kha) [inline src/Lean/Elab/Frontend.lean:82]: `processCommand` and the whole `FrontendM` are not used by Lean itself anymore now but I think it's fine to leave them around as they have some external uses. They just will not benefit from parallelism etc. going forward.

**Pull request merged**